client/state/data-layer/wpcom/wordads/settings/index.js

```diff
@@ -90,13 +90,23 @@ export const handleSaveSuccess = ( { siteId } ) => [
 	} ),
 ];
 
-export const handleSaveFailure = ( { siteId, meta: { settings: previousSettings } } ) => [
+export const handleSaveFailure = ( {
+	siteId,
+	meta: {
+		settings: previousSettings,
+		dataLayer: {
+			error: { error },
+		},
+	},
+} ) => [
 	saveWordadsSettingsFailure( siteId ),
 	updateWordadsSettings( siteId, previousSettings ),
-	errorNotice( translate( 'An unexpected error occurred. Please try again later.' ), {
-		id: `wordads-notice-error-${ siteId }`,
-		duration: 5000,
-	} ),
+	errorNotice(
+		error === 'invalid_paypal'
```

Comment:
should this be a yoda condition?

Reply:
Sounds like it's optional: https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#yoda-conditions